### DISCUSSION TITLE
Heatbed Hotter than Reported

### DISCUSSION THREAD

**Discussion by b-fitzpatrick** [dz0ny/klipper-prusa-mk3s#40] [General]
I was troubleshooting some print issues with a part that has 45° overhangs directly off the bed. It turned out that the heatbed temperature was under-reported by about 10°C at an indicated 90°C (was actually 100°C). I measured the top of the bed with an IR thermometer directly over the thermistor. I re-flashed the stock firmware, and agreement was good, so the hardware is good.

I found a drawing for the MK3S heatbed thermistor in the MINI section of the Prusa GitHub, and it specifies a "Semitec 104NT-4 R025H42G". That's a built-in option in Klipper, but it made the problem a few degrees worse. I ended up getting good agreement by using that thermistor and adding "inline_resistor: 2800".

Anyone else experiencing a similar issue? I would have never known if it weren't for that one print. I do see another discussion about trouble with higher bed temps that may be related.

**Comment by f1yn** [comment]
Thermistors are tricky business, and often when a replacement hotend is purchased it's not guaranteed to use the same thermister and possibility has a different temperature curve.

Did you ever replace the ~hotend~ heatbed (sorry), and if so, was it one from amazon or another marketplace? Would you be able to share a link to it?

Prusa Research is pretty consistent when it comes to not deviating parts, so my best guess would be trying to figure out what termistor you have and go from there.

**Reply by f1yn** [reply]
Klipper relies heavily on knowing the percise thermistor when calibrating PID, while Prusa firmware does a little more work to account for deviations.

Like I said, it's tricky business, but knowing more information will allow me to help.

**Reply by b-fitzpatrick** [reply]
My heatbed is stock. I may have replaced the thermistor due to a broken wire, but if so, it was with a replacement kit from Prusa.

I assume, from the Prusa drawing (https://github.com/prusa3d/Original-Prusa-MINI/blob/master/DOCUMENTATION/ELECTRONICS/MK3s-HB-termistor.pdf) that I have a Semitec 104NT-4 R025H42G. I have a couple more heatbed thermistor replacement kits from Prusa, but I didn't find any markings on the thermistors.

PID tune wasn't the issue. The temperature was stable, just wrong at elevated temperatures. It was correct at room temperature.

I just wonder if everyone else with Klipper on an MK3S has the same issue and hasn't figured it out yet. Or, it there some hardware variance that the Prusa firmware handles? I wasn't able to find anything, but I'm no firmware expert.

**Reply by dz0ny** [reply]
Try asking in Klipper Discord, in general, NTC does not have linear resistance vs temperature so it generally skews after going away from room temperature (25c). Did you do any other modifications, or it's a stock printer?

**Reply by b-fitzpatrick** [reply]
I already asked in the Klipper Discord (https://discord.com/channels/431557959978450984/431559228050636820/1131376510348054569). The printer is stock as far as anything that could possibly be related to this (Dragon HF and some printed bracing).

I'm familiar with how thermistors work and the standard voltage-divider circuit with a pull-up resistor to get a voltage signal. I first went down the path of creating an adc_temperature sensor in Klipper using table values for thermistor "1" in the stock firmware. That behaved similarly wrong. I determined values to shift the ADC values such that there was no change at 25°C and enough for a -10°C change at 90°C, and that worked pretty well. The inline_resistor setting with a standard thermistor worked better and is simpler.

Do any of you have a thermometer suitable for checking your bed temps? They may be higher than you think. I have other mk3s printers that I will check eventually, but they're all still on stock firmware and busy.

**Comment by confuzled-dev** [comment]
I think the issue we were having is the heater bed, not hotend. I have
replaced my hotend heater cartridge but not my bed

**Reply by f1yn** [reply]
Yeah, I corrected my comment right before your reply. Temperature curve and PID are principally the same on the heatbed so hopefully it's just a variant thermistor

**Comment by confuzled-dev** [comment]
An A/B test with a calibrated pyrometer between stock and klipper.

**Reply by b-fitzpatrick** [reply]
Did you try to include an image or something? I don't see anything.
